### PR TITLE
API-12790 Added augment to add state abbreviations to patient records.

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**

### CODE REVIEW

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  static DatamartPatient addStateAbbreviations(Augmentation.Context<DatamartPatient> ctx) {
+    DatamartPatient dm = ctx.resource();
+
+    checkState(!dm.address().isEmpty());
+    var addresses = dm.address();
+
+    for (var address : addresses) {
+      if (address.state() != null) {
+        address.stateAbbreviation(STATE_ABBREVIATIONS.get(address.state()));
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:26]: Are there any existing state values that don't map to an abbreviation?

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return dm;
+  }
+
+  @SneakyThrows
+  static Map<String, String> initStateAbbreviations() {
+    var csvFile =
+        PatientStateAbbreviationAugments.class.getResourceAsStream("us-state-abbreviations.csv");
+    InputStreamReader reader = new InputStreamReader(csvFile);
+    Iterable<CSVRecord> rows = CSVFormat.DEFAULT.withFirstRecordAsHeader().parse(reader);
+    Map<String, String> cvxStateMap = new HashMap<>();
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:39]: cvx?

**Comment by taylorrassmannlibertyits** [comment]
> Are there any existing state values that don't map to an abbreviation?

To my knowledge no.

CC @nmoisan @jtaylor-va @haidong

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java (hunk 1 above)
Comment (jblefkove-va) [reply minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:26]: If not, let's assert here that the result of the lookup is non-null

File: minimart-manager/src/test/resources/gov/va/api/health/minimartmanager/minimart/augments/cdw-state-stateabbreviations.csv [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+VERMONT,VT
+VIRGIN ISLANDS,VI
+VIRGINIA,VA
+WASHINGTON,WA
+WASHINGTON ISC,NULL
+WEST AFRICA,ZWA
+WEST VIRGINIA,WV
+WISCONSIN,WI
+WYOMING,WY
+XXXAMERICAN SAMOA,XXX
```
Comment (jblefkove-va) [inline minimart-manager/src/test/resources/gov/va/api/health/minimartmanager/minimart/augments/cdw-state-stateabbreviations.csv:181]: image

**Comment by jblefkove-va** [comment]
```
{
  "objectType": "Patient",
  "objectVersion": 1,
  "fullIcn": "1000003",
  "ssn": "999-30-3951",
  "name": "Mr. Tobias236 Wolff180",
  ...
  "address": [{
      "street1": "111 MacGyver Viaduct",
      "city": "Anchorage",
      "state": "Alaska",
      "stateAbbreviation": "AK",
      "postalCode": "99501"
    }
  ],
  "contact": [{
      "name": "O'Conner582, Anneliese959",
      "type": "SECONDARY EMERGENCY CONTACT",
      ...
      "address": {
        "street1": "76210 Stehr Causeway",
        "street2": "Suite 315",
        "city": "South Carleetown",
        "state": "Texas",
        "postalCode": "42094"
      }
    }
  ]
}
```

Should contact addresses also have state abbreviation?

**Comment by jblefkove-va** [comment]
To avoid confusion, don't merge this until `ImmunizationManufacturerReferenceAugments` is applied and merged

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    dm.address(addresses);
+    return dm;
+  }
+
+  @SneakyThrows
+  static TreeMap<String, String> initStateAbbreviations() {
+    var csvFile =
+        PatientStateAbbreviationAugments.class.getResourceAsStream(
+            "cdw-state-stateabbreviations.csv");
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:39]: nitpick, but `state-abbreviations`

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    InputStreamReader reader = new InputStreamReader(csvFile);
+    Iterable<CSVRecord> rows = CSVFormat.DEFAULT.withFirstRecordAsHeader().parse(reader);
+    TreeMap<String, String> cvsStateMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:42]: csv

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    var addresses = dm.address();
+
+    for (var address : addresses) {
+      if (address.state() != null) {
+        var abbreviation = STATE_ABBREVIATIONS.get(address.state());
+        checkState(abbreviation != null);
+        address.stateAbbreviation(abbreviation);
+      }
+    }
+    dm.address(addresses);
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:31]: this line is unnecessary; the original list of addresses on the object is modified directly

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+  static DatamartPatient addStateAbbreviations(Augmentation.Context<DatamartPatient> ctx) {
+    DatamartPatient dm = ctx.resource();
+    var addresses = dm.address();
+    checkState(!addresses.isEmpty());
+    for (var address : addresses) {
+      updateAddress(address);
+    }
+    var contacts = dm.contact();
+    if (!contacts.isEmpty()) {
```
Comment (jblefkove-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/augments/PatientStateAbbreviationAugments.java:25]: This check is unnecessary; `dm.contact()` is lazily-initialized and won't be null, and if it's empty, the `for` loop just won't do anything.

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**